### PR TITLE
feat(insights): Wrap sidebar navigation items in `sidebar:navigation-item` hook

### PR DESCRIPTION
**Pull request by gggritso**
Declares a new hook called `sidebar:navigation-item`, and wraps all navigation items in it. Here's the scoop on the hook:

- the hook is a _render prop hook_. It accept a render function child, and passes it some computed properties
- the properties are:
    - `disabled`, which indicates if the child should be "active" or not, whatever that means
    - `AdditionalContent` which supplies some React content to render somewhere alongside the child
    - `Wrapper` which supplies a React wrapper to wrap the child in

The `SidebarItem` component uses this hook by checking the `disabled` property, and if present, removes the `to` on the link (thereby disabling it), rendering the `AdditionalContent` next to the item label, and wrapping _the whole thing_ in `Wrapper`.

The default fallthrough behaviour is that the link is enabled, there's no wrapper, and there's no additional content. i.e., no changes to current behaviour.

Why do all this? We want to be able to have _more control_ over the sidebar items from `getsentry`. We want to:

1. Be able to turn off the entire link item (i.e., disable the link)
2. Wrap the whole link in an upsell dialog
3. Pass it some extra content to render (i.e., the Business icon)

The existing `sidebar:item-label` hook doesn't help us, because it doesn't provide a mechanism for wrapping, or turning off the link! See https://github.com/getsentry/getsentry/pull/14127 for the corresponding `getsentry` hook behaviour

### CODE REVIEW

File: static/app/types/hooks.tsx [hunk 1]
```diff
@@ -646,6 +647,20 @@ type InviteButtonCustomizationHook = () => React.ComponentType<{
...
 }>;
 
+/**
+ * Sidebar navigation item customization allows passing render props to disable
+ * the link, wrap it in an upsell modal, and give it some additional content
+ * (e.g., a Business Icon) to render.
+ */
+type SidebarNavigationItemHook = () => React.ComponentType<{
+  children: (opts: {
+    AdditionalContent: React.ReactElement | null;
```
Comment (evanpurkhiser) [inline static/app/types/hooks.tsx:657]: Why is this capitalized

Comment (gggritso) [reply static/app/types/hooks.tsx:657]: My bad, it was originally also a renderable, like `Wrapper`! I changed it, but didn't lowercase. I'll update 👍🏻

File: static/app/components/sidebar/sidebarItem.tsx [hunk 2]
```diff
@@ -299,6 +308,16 @@ export function isItemActive(
   );
 }
 
+const SidebarNavigationItemHook = HookOrDefault({
+  hookName: 'sidebar:navigation-item',
+  defaultComponent: ({children}) =>
+    children({
+      disabled: false,
+      AdditionalContent: <Fragment />,
```
Comment (evanpurkhiser) [inline static/app/components/sidebar/sidebarItem.tsx:316]: Can just be `null`?

File: static/app/types/hooks.tsx [hunk 3]
```diff
@@ -646,6 +647,20 @@ type InviteButtonCustomizationHook = () => React.ComponentType<{
   organization: Organization;
 }>;
 
+/**
+ * Sidebar navigation item customization allows passing render props to disable
+ * the link, wrap it in an upsell modal, and give it some additional content
+ * (e.g., a Business Icon) to render.
+ */
```
Comment (evanpurkhiser) [inline static/app/types/hooks.tsx:654]: maybe let's have a TODO above this that sates how we could replace the label hook with this one potentially

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/components/sidebar/sidebarItem.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/components/sidebar/sidebarItem.tsx:316]: It can! Will change

**Pull request merged**